src/WebDocs/wwwroot/app/shared/header.html

```diff
@@ -1,8 +1,32 @@
 ﻿<!DOCTYPE html>
 <script src="/drapo.js"></script>
 <div>
-    <div>
-        <img src="../../img/drapo.png" class="dHeaderImage"/>
-        <span>Drapo Docs</span>
+    <div class="header-container">
+        <div class="header-left">
+            <img src="../../img/drapo.png" class="dHeaderImage"/>
+            <span>Drapo Docs</span>
+        </div>
+        <div class="header-right">
+            <div class="search-container">
+                <div d-dataKey="searchQuery" d-dataType="value"></div>
+                <div d-dataKey="searchResults" d-dataUrlGet="/api/Search/Search?query={{searchQuery}}" d-dataUrlGetOnChange="searchQuery"></div>
+                <div d-dataKey="showSearchResults" d-dataType="value" d-dataValue="false"></div>
+                <input type="text" 
+                       class="search-input" 
+                       placeholder="Search documentation..." 
+                       d-model="{{searchQuery}}" 
+                       d-model-event="keyup"
+                       d-on-focus="IF(LENGTH({{searchQuery}}) >= 3, UpdateData(showSearchResults,true))"
+                       d-on-blur="Async(UpdateData(showSearchResults,false),200)" />
+                <div class="search-results" d-if="{{showSearchResults}} && LENGTH({{searchResults}}) > 0 && LENGTH({{searchQuery}}) >= 3">
```

Comment:
there is no drapo function called LENGTH. in this specific case you can use {{searchResults.length}}